### ISSUE TITLE
An error occurred while migrating Craft CMS from 3.4.5 -> 3.4.6.1

### ISSUE THREAD

**Issue by khalwat** [craftcms/cms#5687]
### Description

Deployed changes via CI, the following packages were updated:

```
  - Updating aws/aws-sdk-php (3.133.15 => 3.133.18): Downloading (100%)
  - Updating craftcms/cms (3.4.5 => 3.4.6.1): Downloading (connecting...)Downloading (100%)         
```

Then migrations were run:
```
@php craft migrate/all
```

Greeted with:
```
Yii Migration Tool (based on Yii v2.0.32)

    > truncate table {{%templatecachequeries}} ... done (time: 0.011s)
    > create index elements_archived_dateDeleted_draftId_revisionId_idx on {{%elements}} (archived,dateDeleted,draftId,revisionId) ... done (time: 0.008s)
Exception 'craft\errors\MigrateException' with message 'An error occurred while migrating Craft CMS.'

in /home/forge/devmode.fm/cms/vendor/craftcms/cms/src/services/Updates.php:241

Stack trace:
#0 /home/forge/devmode.fm/cms/vendor/craftcms/cms/src/console/controllers/MigrateController.php(259): craft\services\Updates->runMigrations(Array)
#1 [internal function]: craft\console\controllers\MigrateController->actionAll()
#2 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#3 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#4 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction('all', Array)
#5 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#6 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#7 /home/forge/devmode.fm/cms/vendor/craftcms/cms/src/console/Application.php(87): yii\console\Application->runAction('migrate/all', Array)
#8 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#9 /home/forge/devmode.fm/cms/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#10 /home/forge/devmode.fm/cms/craft(22): yii\base\Application->run()
#11 {main}
Script @php craft migrate/all handling the craft-update event returned with error code 1

```

### Additional info

- Craft version: 3.4.5
- PHP version: 7.1.33
- Database driver & version: Postgres 9.5.19
- Plugins & versions: n/a

**Comment by khalwat** [comment]
mmmmm. Might have been a CI pipeline issue; closing unless other information is found

**Comment by niektenhoopen** [comment]
@khalwat Did you find more information by any chance?

I am running into the same error when upgrading from Craft 3.4.5 to 3.4.9:
```
Yii Migration Tool (based on Yii v2.0.32)

    > truncate table {{%templatecachequeries}} ... done (time: 0.203s)
    > create index elements_archived_dateDeleted_draftId_revisionId_idx on {{%elements}} (archived,dateDeleted,draftId,revisionId) ... done (time: 0.430s)
    > alter column message in table {{%deprecationerrors}} to text ... done (time: 0.259s)
Exception 'craft\errors\MigrateException' with message 'An error occurred while migrating Craft CMS.'

in /home/forge/*********/releases/20200301130715/vendor/craftcms/cms/src/services/Updates.php:241
```

**Comment by niektenhoopen** [comment]
Ah, I think I found it. `allowAdminChanges` is false on my server and I forgot to commit the project.yaml changes. Therefore it wanted to run migrations on the server and update the project.yaml. In the console.log is a clearer error message that states that migrations are unable to run because the project.yaml is read-only. Would have been nicer to see that error message in the console while running the migrations and not 'hidden' in the console.log.

**Comment by mattandrews** [comment]
I had the same error recently and thought I'd warn anybody who had the same cause: we're using dependabot to automatically upgrade our Composer dependencies, but when it changes the Craft version, it doesn't update `project.yaml` (as @niektenhoopen suggests above), so my deploy failed for the same reasons (eg. `allowAdminChanges` being false in production). I made the upgrade manually locally (eg. without dependabot) and commited the `project.yaml` file and everything worked. 

TL;DR – don't let bots upgrade Craft for you!
